crates/astria-sequencer/src/accounts/query.rs

```diff
@@ -100,6 +102,13 @@ pub(crate) async fn balance_request(
             };
         }
     };
+
+    // Unstable sort does not allocate auxillary memory and is typically faster. This utilizes a
+    // custom `Ord` implementation which sorts by `Denom` in alphabetical order. The comparison
+    // should never return `Ordering::Equal` since `Denom` is a unique identifier, hence the
+    // unstable sort will be deterministic.
+    balances.sort_unstable();
```

Comment:
Instead of using the problematic `Ord` on `AssetBalance`, use https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable_by_key:
```rust
balances.sort_unstable_by_key(asset_balance.denom());
```

Reply:
This method requires cloning the denom since it doesn't implement `Copy`. Is there a different way you'd like to accomplish this without extra allocation? Aside from that, my best suggestion would be to just fix the `Ord` impl.

Reply:
Why do you need to clone? There is a blanket `impl Ord for &A where A: Ord`. So implement `AssetBalance::denom(&self) -> &Denom`, so you can call `balances.sort_unstable_by_key(AssetBalance::denom)`.

Reply:
Implemented changes we discussed offline in 0af3bcf.